### DISCUSSION TITLE
UID is not necessary, also strip the blank lines.

### DISCUSSION THREAD

**Discussion by jablka** [ics-py/ics-py#390] [Ideas]
hello, 
when exporting to `.ics` file, 
this module creates a unique **'UID'** for each event in the calendar, and it **looks like a random email address**.
(it is generated in `utils.py` function `uid_gen()` )
I think it's unnecessary and confusing... (I was able to upload the ics file to my google calendar without those UID lines in each event.)
Shouldn't we get rid of it?

Also I would suggest a bit nicer output. Currently after each line follows a blank line (in `.ics` file).
Either strip those blank lines altogether, 
or (better, but more laborous) seperate with blank lines just some logical context (like file header and particular events).

Overall it's a fine package, 
thank you for it.

**Comment by N-Coder** [comment]
The `UID` property is [compulsory](https://www.kanzaki.com/docs/ical/uid.html) for valid `VEVENT`s, so simply dropping it would yield an invalid ics file. The link als provides guidelines on how this UID should be generated.

Regarding the empty line, how are you serializing the calendar? This sounds like a issue with writing individual lines (and inserting linebreaks between them) while the lines already end with a line break.

**Reply by jablka** [reply]
just by following the documentation:
```python
from ics import Calendar, Event
c = Calendar()
e = Event("My cool event", begin='2023-07-08 00:00:00')
c.events.add(e)
with open('my3.ics', 'w') as f:
    f.writelines(c.serialize_iter())
# And it's done !
```

will produce:
```
BEGIN:VCALENDAR

VERSION:2.0

PRODID:ics.py - http://git.io/lLljaA

BEGIN:VEVENT

DTSTART:20230708T000000Z

SUMMARY:My cool event

UID:99903d8f-0ddc-4604-9bd7-fc571afe5be4@9990.org

END:VEVENT

END:VCALENDAR
```

**Reply by jablka** [reply]
aha, I see...
`serialize functions` separate the lines by `\r\n`
but maybe just `\n` would be enough?
(btw. I'm on Windows)

**Reply by jablka** [reply]
ok, I solved it for myself,
this works:
```python
s = c.serialize() # s is a string
s = s.replace('\r\n','\n') # replacing '\r\n' with '\n'
with open('output.ics', mode='w', encoding='utf-8') as f:
    f.write(s) 
```

**Reply by N-Coder** [reply]
FYI, the standard [mandates](https://icalendar.org/iCalendar-RFC-5545/3-1-content-lines.html) the line separator to be `\r\n` on all systems.
